crates/sui-transactional-test-runner/src/lib.rs

```diff
@@ -9,11 +9,154 @@ pub mod test_adapter;
 
 use move_transactional_test_runner::framework::run_test_impl;
 use std::path::Path;
+use sui_types::storage::ObjectStore;
 use test_adapter::{SuiTestAdapter, PRE_COMPILED};
 
+use std::sync::Arc;
+use sui_core::authority::authority_test_utils::send_and_confirm_transaction_with_execution_error;
+use sui_core::authority::AuthorityState;
+use sui_json_rpc_types::DevInspectResults;
+use sui_json_rpc_types::EventFilter;
+use sui_json_rpc_types::SuiEvent;
+use sui_storage::key_value_store::TransactionKeyValueStore;
+use sui_types::base_types::ObjectID;
+use sui_types::base_types::SuiAddress;
+use sui_types::base_types::VersionNumber;
+use sui_types::effects::TransactionEffects;
+use sui_types::error::ExecutionError;
+use sui_types::error::SuiError;
+use sui_types::error::SuiResult;
+use sui_types::event::EventID;
+use sui_types::messages_checkpoint::VerifiedCheckpoint;
+use sui_types::object::Object;
+use sui_types::transaction::Transaction;
+use sui_types::transaction::TransactionDataAPI;
+use sui_types::transaction::TransactionKind;
+
 #[cfg_attr(not(msim), tokio::main)]
 #[cfg_attr(msim, msim::main)]
 pub async fn run_test(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
     run_test_impl::<SuiTestAdapter>(path, Some(&*PRE_COMPILED)).await?;
     Ok(())
 }
+
+pub struct ValidatorWithFullnode {
+    pub validator: Arc<AuthorityState>,
+    pub fullnode: Arc<AuthorityState>,
+    pub kv_store: Arc<TransactionKeyValueStore>,
+}
+
+#[allow(unused_variables)]
+/// TODO: better name?
+#[async_trait::async_trait]
+pub trait TransactionalAdapterTestable: Send + Sync + ObjectStore {
```

Comment:
why not just TransactionalAdapter drop the Testable

Reply:
+1